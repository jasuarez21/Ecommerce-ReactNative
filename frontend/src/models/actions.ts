import { Moto } from "./moto";
import { User } from "./user";

export type MotoActions = 
    | {type: 'GET_MOTOS'; data: Array<Moto>}
    | {type: 'ADD_TO_CART'; moto: Moto}
    | {type: 'LOAD_CART';}
    | {type: 'DELETE_CART'; moto: Moto}
    | {type: 'LOG_IN'; data: User}
    | {type: 'LOG_OUT'; data: User}
    | {type: 'SIGN_UP'; data: User}
    | {type: 'UPDATE_CART'}
    | {type: 'DELETE_ACCOUNT'}
    | {type: 'ERROR'}