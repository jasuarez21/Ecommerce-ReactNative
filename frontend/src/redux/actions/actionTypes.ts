const actionTypes = {
  GET_MOTOS: 'GET_MOTOS',
  LOG_IN: 'LOG_IN',
  LOG_OUT: 'LOG_OUT',
  SIGN_UP: 'SIGN_UP',
  ADD_TO_CART: 'ADD_TO_CART',
  LOAD_CART: 'LOAD_CART',
  DELETE_CART: 'DELETE_CART',
  UPDATE_CART: 'UPDATE_CART',
  SAVE_PRODUCT: 'SAVE_PRODUCT',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  ERROR: 'ERROR'
};

export default actionTypes;
