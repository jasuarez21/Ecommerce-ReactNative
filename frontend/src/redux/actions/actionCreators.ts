import { User } from './../../models/user';
import axios from 'axios';
import {URL_MOTOS, URL_LOGIN, URL_LOGOUT, URL_SIGNUP, URL_DELETE_ACCOUNT, URL_SAVE_PRODUCTS} from '@env';
import actionTypes from './actionTypes';
import {Moto} from '../../models/moto';
import {Dispatch} from 'redux';
import {MotoActions} from '../../models/actions';
import { Order } from '../../models/order';

export function GetMotos() {
  return async (dispatch: Dispatch) => {
      try {   
      const {data} = await axios.get(URL_MOTOS);
      dispatch({
        type: actionTypes.GET_MOTOS,
        data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}



export function login(email: string, password: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGIN, {email, password});
        dispatch({
          type: actionTypes.LOG_IN,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function logout() {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_LOGOUT);
        dispatch({
          type: actionTypes.LOG_OUT,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}


export function signUp(email: string, password: string, alias: string, avatar: string) {
  return async (dispatch: Dispatch) => {
      try {   
        const {data} = await axios.post(URL_SIGNUP, {email, password, alias, avatar});
        dispatch({
          type: actionTypes.SIGN_UP,
          data,
        });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function addToCart(moto: Moto): MotoActions {
  return {
    type: 'ADD_TO_CART',
    moto
  };
}

export function loadCart(): MotoActions {
  return {
    type: 'LOAD_CART'
  };
}

export function deleteFromCart(moto: Moto): MotoActions {
  return {
    type: 'DELETE_CART',
    moto,
  };
}

export function reloadCart(moto: Array<Moto>){
  return async (dispatch: Dispatch) => {
    try {   
      for(let i = 0; i < moto.length; i++){
        await axios.post(`${URL_MOTOS}/${moto[i]._id}`, moto[i]);
        dispatch({
          type: actionTypes.UPDATE_CART
        });
      }
  } catch (error) {
    dispatch({
      type: actionTypes.ERROR,
    });
  }
}
}

export function saveOrder(user: User, cart: Order, token: string){
  user.cart.push(cart);
  const config = { 
    headers: {
        authorization: `Bearer ${token}`,
    }
  }
    return async (dispatch: Dispatch) => {
      try {   
          await axios.post(URL_SAVE_PRODUCTS, {_id: user._id, cart: user.cart}, config);
          dispatch({
            type: actionTypes.SAVE_PRODUCT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}

export function deleteAccount(user: User, token: string){
    return async (dispatch: Dispatch) => {
      try {   
          await axios.delete(URL_DELETE_ACCOUNT, {headers: {
              authorization: `Bearer ${token}`,
          },
              data:{_id: user._id}
          });
          dispatch({
            type: actionTypes.DELETE_ACCOUNT
          });
    } catch (error) {
      dispatch({
        type: actionTypes.ERROR,
      });
    }
  }
}