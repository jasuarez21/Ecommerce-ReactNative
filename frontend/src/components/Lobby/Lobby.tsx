import React, {useState} from 'react';
import {RootState} from '../../redux/store/index'
import {Sugestions} from '../Sugestions/Sugestions';
import {StackScreenProps} from '@react-navigation/stack';
import {useSelector, useDispatch} from 'react-redux';
import {logout} from '../../redux/actions/actionCreators';
import {TouchableHighlight, View, Text, TextInput, SafeAreaView, ScrollView,  StyleSheet, Image} from 'react-native';


interface Props extends StackScreenProps<any, any> {}

const Lobby = ({navigation}: Props) => {
  const user = useSelector((store:RootState) => store.user);
  const dispatch = useDispatch();
  let [motoSearch, setMotoSearch] = useState('');
  return (
    <View style={styles.backgroundPage}>
      <SafeAreaView>
        <ScrollView>
     {user.token === undefined ?
    (
      <>
      <Image
        style={styles.logo}
        source={{
          uri: 'https://i.postimg.cc/FF2Y5GJd/logo-small.png',
        }}
      />
      <Text style={styles.userUndefined}>Bienvenid@! Registrate, porfavor!</Text>
      <View style={styles.containerSearch}>
        <TextInput
          style={styles.input}
          testID='searchInput'
          placeholder="Que buscas?"
          autoCapitalize="none"
          onChangeText={(motoType) => {setMotoSearch(motoType)}}
          defaultValue={motoSearch}
        />
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: motoSearch})}>
          <Image
            testID='logout'
            style={styles.icon}
            source={{
              uri: 'https://i.postimg.cc/dV8GDfQL/icons8-search-50.png',
            }}
          />
          </TouchableHighlight>
      </View>
      </>
    ) : (
      <>
      <View style={styles.welcome}>
        <Image 
          style={styles.logoLogged}
          source={{
            uri: 'https://i.postimg.cc/j2236kSp/logo-small.png'
          }}
        />
        <Text style={styles.user}>Bienvenid@, {user?.user?.alias}!</Text>
      <TouchableHighlight
          onPress={() => dispatch(logout())}>
           <Image
                testID='logout2'
                style={styles.logout}
                source={{
                  uri: 'https://i.postimg.cc/ZnpJHx1s/icons8-logout-rounded-left-96.png',
                }}
              />
        </TouchableHighlight>
      </View>
      <View style={styles.containerSearch}>
        <TextInput
          testID='searchInput'
          style={styles.input}
          placeholder="Que buscas?"
          autoCapitalize="none"
          onChangeText={(motoType) => {setMotoSearch(motoType)}}
          defaultValue={motoSearch}
        />
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: motoSearch})}>
          <Image
            testID= 'navigateButton'
            style={styles.icon}
            source={{
              uri: 'https://i.postimg.cc/dV8GDfQL/icons8-search-50.png',
            }}
          />
          </TouchableHighlight>
      </View>
      <Sugestions />
      </>
    )
  }
        <Text style={styles.category}>CATEGORIAS</Text>
      <View style={styles.lobby}>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'scooter'})}>
          <Image
            testID='scooter'
            style={styles.foto}
            source={{
              uri: 'https://i.postimg.cc/HntMGvdq/pcr2.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'ofertas'})}>
          <Image
            testID='offer'
            style={styles.foto}
            source={{
              uri: 'https://www.retif.es/media/image/400x/cartel-ofertas-horizontal-86x20-cm-blanco-rojo_01.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'racing'})}>
          <Image
            testID='racing'
            style={styles.foto}
            source={{
              uri: 'https://i.postimg.cc/90c4BPZJ/cbr3.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'vespa'})}>
          <Image
            testID='vespa'
            style={styles.foto}
            source={{
              uri: 'https://i.postimg.cc/0yjw3C34/vespa4.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'custom'})}>
          <Image
            testID='custom'
            style={styles.foto}
            source={{
              uri: 'https://i.postimg.cc/6pbd5Pnw/m109R3.jpg',
            }}
          />
        </TouchableHighlight>
        <TouchableHighlight
          onPress={() => navigation.navigate('List', {typeMoto: 'casco'})}>
          <Image
            testID='casco'
            style={styles.foto}
            source={{
              uri: 'https://www.motocat.net/3656-thickbox/casco-integral-mt-imola-ii-solid-matt-black.jpg',
            }}
          />
        </TouchableHighlight>
      </View>
        </ScrollView>
      </SafeAreaView>
    </View>
  );
};

const styles = StyleSheet.create({
  foto: {
    width: 175,
    height: 110,
    borderRadius: 25,
    marginTop: 20,
  },
  logoLogged: {
    height: 70,
    width: 60,
    position: 'absolute',
    marginBottom: 50,
    marginLeft: 20,
    alignSelf: 'center'
  },
  icon: {
    height: 30,
    width: 30,
    marginTop: 15,
    marginLeft: 5
  },
  welcome: {
    backgroundColor: '#464646',
    height: 70,
    width: 450,
    textAlign: 'center',
    display: 'flex',
    padding: 10,
    flexDirection: 'row',
    marginTop: 15
  },
  category:{
    color: 'white',
    fontSize: 18,
    marginTop: 15,
    marginLeft: 20,
    borderBottomWidth: 1,
    width: 370,
    padding: 4,
    borderBottomColor: '#FFB800'
  },
  input: {
    backgroundColor: 'white',
    width: 250,
    height: 35,
    marginTop: 10,
    marginLeft: 20,
    borderRadius: 25,
  },
  containerSearch: {
    display: 'flex',
    flexDirection: 'row',
    alignSelf: 'center',
    marginTop: 15
  },
  user: {
    color: 'white',
    fontSize: 16,
    marginLeft: 200,
    alignSelf: 'center'
  },
  userUndefined: {
    color: 'white',
    marginTop: 10,
    marginLeft: 160,
    fontSize: 16
  },
  logout: {
    width: 25,
    height: 25,
    marginTop: 10,
    marginLeft: 10
  },
  logo: {
    width: 300,
    height: 42,
    marginTop: 15,
    marginLeft: 10
  },
  backgroundPage: {
    backgroundColor: 'black',
  },
  lobby: {
    backgroundColor: 'black',
    height: 600,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    marginTop: 10,
  },
});

export default Lobby;
