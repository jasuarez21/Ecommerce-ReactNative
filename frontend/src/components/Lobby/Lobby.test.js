import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/actionCreators';
import { Provider } from 'react-redux';
import Lobby from './Lobby';

jest.mock('../../redux/actions/actionCreators');

describe('When invoked a Lobby component', () => {
    const mockStore = ConfigureStore();
    const store = mockStore({
        user: [{
              _id: '2',
              user : {
                  alias: 'Anna',
                  email: 'annita',
                  password: 'annaaaza',
                  avatar: 'img.com'
              }
          }
        ]
    });
    let navigation;
    beforeEach(() => {
        navigation = {
            navigate: jest.fn()
        };
      });

    test('Should render the lobby of page', () => {
        const lobby = render(<Provider store={store}><Lobby navigation={navigation} /></Provider>)
        expect(lobby).toMatchSnapshot();
    })
})

describe('When image button is pressed', () => {
    const mockStore = ConfigureStore();
    const store = mockStore({
        user: [{
              _id: '2',
              user : {
                  alias: 'Anna',
                  email: 'annita',
                  password: 'annaaaza',
                  avatar: 'img.com'
              }
          }
        ]
    });
    let navigation;
    beforeEach(() => {
        jest.spyOn(actions, 'logout').mockReturnValueOnce({ type: '' });
        dispatch = jest.fn();
        navigation = {
            navigate: jest.fn()
        };
      });
    test('Then navigation.navigate scooter is invoked', () => {
      const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
      const image = getByTestId('scooter');
      fireEvent.press(image);
      expect(navigation.navigate).toHaveBeenCalled();
    });
    test('Then navigation.navigate offer is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('offer');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then navigation.navigate racing is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('racing');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then navigation.navigate custom is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('custom');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then navigation.navigate vespa is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('vespa');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then navigation.navigate casco is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('casco');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then logout is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('logout');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
      test('Then setMotoSearch is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby /></Provider>);
        const searchInput = getByTestId('searchInput');
        const setMotoSearch = jest.fn()
        fireEvent.changeText(searchInput, setMotoSearch);
       expect(setMotoSearch).toHaveBeenCalled();
      });
  });

  describe('When image button is pressed', () => {
    const mockStore = ConfigureStore();
    const store = mockStore({
        user:{ 
              token : 'baibfbaibaka',
              _id: '2',
              user : {
                  alias: 'Anna',
                  email: 'annita',
                  password: 'annaaaza',
                  avatar: 'img.com'
              }
          }
    });
    let navigation;
    beforeEach(() => {
        jest.spyOn(actions, 'logout').mockReturnValueOnce({ type: '' });
        dispatch = jest.fn();
        navigation = {
            navigate: jest.fn()
        };
      });
      test('Then logout func is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('logout2');
        fireEvent.press(image);
        expect(actions.logout).toHaveBeenCalled();
      });
      test('Then setMotoSearch is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby /></Provider>);
        const searchInput = getByTestId('searchInput');
        const setMotoSearch = jest.fn()
        fireEvent.changeText(searchInput, setMotoSearch);
       expect(setMotoSearch).toHaveBeenCalled();
      });
      test('Then navigation.navigate is invoked', () => {
        const { getByTestId } = render(<Provider store={store}><Lobby navigation={navigation}/></Provider>);
        const image = getByTestId('navigateButton');
        fireEvent.press(image);
        expect(navigation.navigate).toHaveBeenCalled();
      });
  });