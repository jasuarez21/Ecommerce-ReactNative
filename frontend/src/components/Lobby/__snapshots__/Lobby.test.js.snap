// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Lobby component Should render the lobby of page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <RCTScrollView>
      <View>
        <Image
          source={
            Object {
              "uri": "https://i.postimg.cc/FF2Y5GJd/logo-small.png",
            }
          }
          style={
            Object {
              "height": 42,
              "marginLeft": 10,
              "marginTop": 15,
              "width": 300,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "marginLeft": 160,
              "marginTop": 10,
            }
          }
        >
          Bienvenid@! Registrate, porfavor!
        </Text>
        <View
          style={
            Object {
              "alignSelf": "center",
              "display": "flex",
              "flexDirection": "row",
              "marginTop": 15,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            defaultValue=""
            onChangeText={[Function]}
            placeholder="Que buscas?"
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 25,
                "height": 35,
                "marginLeft": 20,
                "marginTop": 10,
                "width": 250,
              }
            }
            testID="searchInput"
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.postimg.cc/dV8GDfQL/icons8-search-50.png",
                }
              }
              style={
                Object {
                  "height": 30,
                  "marginLeft": 5,
                  "marginTop": 15,
                  "width": 30,
                }
              }
              testID="logout"
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "borderBottomColor": "#FFB800",
              "borderBottomWidth": 1,
              "color": "white",
              "fontSize": 18,
              "marginLeft": 20,
              "marginTop": 15,
              "padding": 4,
              "width": 370,
            }
          }
        >
          CATEGORIAS
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 600,
              "justifyContent": "space-around",
              "marginTop": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.postimg.cc/HntMGvdq/pcr2.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="scooter"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://www.retif.es/media/image/400x/cartel-ofertas-horizontal-86x20-cm-blanco-rojo_01.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="offer"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.postimg.cc/90c4BPZJ/cbr3.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="racing"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.postimg.cc/0yjw3C34/vespa4.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="vespa"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.postimg.cc/6pbd5Pnw/m109R3.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="custom"
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              source={
                Object {
                  "uri": "https://www.motocat.net/3656-thickbox/casco-integral-mt-imola-ii-solid-matt-black.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 25,
                  "height": 110,
                  "marginTop": 20,
                  "width": 175,
                }
              }
              testID="casco"
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
